storage/stream_flash: Add API to query buffered data size

SyverHaraldsen opened the pull request:
Add a wrapper function to read the number of bytes currently buffered and pending for the next flash write operation.

Signed-off-by: Syver Haraldsen <syver.haraldsen@nordicsemi.no>

maass-hamburg commented:
making it inline and only in the header?

de-nordic reviewed (changes requested):
Fine with the change.
Missing test case for added function.

de-nordic reviewed (changes requested):
Can you move the test into separate commit?

jtguggedal commented:
I'm a bit curious, why would we want the test in a separate commit? Personally, I find it convenient when an API and example of its use is added in the same commit. Also may make bisecting simpler when API and test are changed atomically.

SyverHaraldsen committed:
storage/stream_flash: Add API to query buffered data size

add a wrapper function to read the number of bytes currently buffered
and pending for the next flash write operation.

Signed-off-by: Syver Haraldsen <syver.haraldsen@nordicsemi.no>

SyverHaraldsen committed:
storage/stream_flash: Add test for buffered data size API

Add a test for the new API to query the buffered data size in the
stream_flash subsystem.

Signed-off-by: Syver Haraldsen <syver.haraldsen@nordicsemi.no>

jtguggedal commented:
Interesting, I was not aware that was enforced in Zephyr.

SyverHaraldsen commented:
No problem, updated the commits.
